Indev to Master

HoverEpic opened the pull request:
Changelog
- Update to 1.2.7 (protocol 160)
- Add more blocks
- Add more blocks to translator (some data values always missing)
- Working animals and monsters entities (spawn / despawn /move)
- Block place / break works in creative mode
- Fix yaw / pitch at login
- Fix player permissions (set to member)
- Send language during login
- format a lot of classes to fit the coding style

robske110 commented:
First time i coded (that was not really coding) java :P

DefinitlyEvil commented:
Nice work, thanks!

HoverEpic committed:
send MC|Brand & client settings (experimental)

HoverEpic committed:
Merge branch 'master' into client-connect-improvments

DefinitlyEvil committed:
Merge branch 'master' into client-connect-improvments

HoverEpic committed:
Add inventory transaction system (wip)

HoverEpic committed:
remove unused inventory transaction case

HoverEpic committed:
add packets

HoverEpic committed:
add actions to interact packet

HoverEpic committed:
fix bad name in adventure settings packet

HoverEpic committed:
add recursivity in debug tool

HoverEpic committed:
add new packets to protocol

HoverEpic committed:
update inventory transaction packet

HoverEpic committed:
remove "UPSTREAMING >> bla bla" message

HoverEpic committed:
disable error in chunk translator

HoverEpic committed:
add sound translator (not tested)

HoverEpic committed:
add animation translator (not tested)

HoverEpic committed:
add pick block request translator (not tested)

HoverEpic committed:
add sond event packet translator : client side (not tested)

HoverEpic committed:
work on inventory transaction translator (not working)

HoverEpic committed:
add getByLocalEID in entity cache

HoverEpic committed:
start working on interact packet translator

HoverEpic committed:
add new translators

HoverEpic committed:
change in intem block translator (move byte in translator)

HoverEpic committed:
add clientbound edventure settings packet translator

HoverEpic committed:
fix adventure settings packet, error in set flag method

HoverEpic committed:
fix spawn pitch yaw in offline mode, fix player permission on join

Creeperface01 committed:
Fix block metadata

Creeperface01 committed:
ops

Creeperface01 committed:
Ops (reader read console output so that was the reason..)

HoverEpic committed:
Merge branch 'master' of https://github.com/Creeperface01/DragonProxy into Creeperface01-master

HoverEpic committed:
Merge branch 'Creeperface01-master' into indev

HoverEpic committed:
Merge branch 'client-connect-improvments' of https://github.com/HoverEpic/DragonProxy into HoverEpic-client-connect-improvments

HoverEpic committed:
Merge branch 'HoverEpic-client-connect-improvments' into indev

lukeeey committed:
Minor refactor

lukeeey committed:
Fix incorrect version

Creeperface01 committed:
Fix block metadata

HoverEpic committed:
Merge branch 'Creeperface01-patch-1' into indev

# Conflicts:
#	proxy/src/main/java/org/dragonet/proxy/network/translator/pc/PCMultiChunkDataPacketTranslator.java

HoverEpic committed:
Update protocole to 1.2.7

HoverEpic committed:
Update ItemBlockTranslator :
- set swap items ids
- add dropper translator
- rewrite simple NBT translator

HoverEpic committed:
Set NBT toString() (human can read NBT)

HoverEpic committed:
Add block entity translation (WIP)

HoverEpic committed:
Fix typo in DebugTools

HoverEpic committed:
Merge branch 'master' into indev

HoverEpic committed:
Fix bad import after merge

HoverEpic committed:
Add BlockEntities translate methode (WIP)

HoverEpic committed:
Add BlockEntities translate methode (WIP)

HoverEpic committed:
Add player hotbar packet

lukeeey committed:
Fix brace style

HoverEpic committed:
Add BlocFace traslation

HoverEpic committed:
remove buggy debug

HoverEpic committed:
add equals() method to BlockPosition

HoverEpic committed:
Implements block place / break / drop (tested only in creative)

lukeeey commented on proxy/src/main/java/org/dragonet/proxy/network/translator/ItemBlockTranslator.java:
```diff
@@ -322,6 +323,11 @@ public static ItemStack translateToPC(Slot slot) {
         return item;
     }
+    
+    public static BlockFace translateToPC(int face)
```
Omfg stop using your own brace style

HoverEpic committed:
add encoding datas to pom

lukeeey commented on proxy/src/main/java/org/dragonet/proxy/network/cache/EntityCache.java:
```diff
@@ -67,6 +67,10 @@ public CachedEntity getClientEntity() {
 	}
 
 	public CachedEntity getByRemoteEID(long eid) {
+            if (!mapRemoteToClient.containsKey(eid))
```
Either fix your formatting and brace style or don't commit at all

HoverEpic replied:
file reformated

lukeeey commented on proxy/src/main/java/org/dragonet/proxy/network/translator/pc/PCPlayerPositionRotationPacketTranslator.java:
```diff
@@ -68,6 +75,15 @@ public PCPlayerPositionRotationPacketTranslator() {
+                        
+                        ByteArrayDataOutput out = ByteStreams.newDataOutput();
+                        out.writeUTF("DragonProxy");
+                        ClientPluginMessagePacket ClientPluginMessagePacket = new ClientPluginMessagePacket("MC|Brand" , out.toByteArray());
```
Formatting again.

lukeeey commented on proxy/src/main/java/org/dragonet/proxy/utilities/LoginChainDecoder.java:
```diff
@@ -57,6 +58,8 @@ public void decode() {
 		// client data
 		clientData = decodeToken(new String(clientDataJWT, StandardCharsets.UTF_8));
+                
+                language = clientData.has("LanguageCode") ? clientData.get("LanguageCode").getAsString() : "en_US";
```
omfg

lukeeey commented on proxy/src/main/java/org/dragonet/proxy/network/translator/pc/PCAnimationPacketTranslator.java:
```diff
@@ -0,0 +1,46 @@
+
+	// public
+	public PEPacket[] translate(UpstreamSession session, ServerEntityAnimationPacket packet) {
+            
```
Are you serious? Don't copy entire classes from other projects.

lukeeey commented on proxy/src/main/java/org/dragonet/proxy/network/translator/pc/PCSoundEventPacketTranslator.java:
```diff
@@ -0,0 +1,64 @@
+        pk.position = new Vector3F((float)packet.getX(), (float)packet.getY(), (float)packet.getZ());
+        pk.pitch = (int)packet.getPitch();
+        switch(packet.getSound())
+        {
```
...

lukeeey commented on proxy/src/main/java/org/dragonet/proxy/network/translator/pe/PEInteractPacketTranslator.java:
```diff
@@ -17,23 +17,35 @@
-	}
+    // public
+    public Packet[] translate(UpstreamSession session, InteractPacket packet) {
+        CachedEntity e = session.getEntityCache().getByLocalEID(packet.targetRtid);
```
Finally someone who knows how to use the TAB button!

HoverEpic replied:
For explanations : the entire project is coded with `\t`, wich is horrible.
I'm replacing all by four spaces tabs when I hardly work on a class.
I'll do the conversion on all classes i pass now.

lukeeey reviewed (changes requested):
Ugh

HoverEpic committed:
add Position to BlockPosition

HoverEpic committed:
TranslatedChatCompenent WIP

HoverEpic committed:
add non Item entity Objects in cache

HoverEpic committed:
implements some data values for entity metadatas

HoverEpic committed:
fix items / armorstands spawning

HoverEpic committed:
add entity link packet to protocole

HoverEpic committed:
add entity set passenger translator (WIP)

HoverEpic committed:
reformate some classes (\t to 4 spaces)

HoverEpic committed:
Reorganize project :
- move proxy project to root
- move Bedrock game data to data dir
- clean some \t in translator classes
- remove unnecessary comment (vars, constructor, public, private)
- reformate some classes to fit the project coding style

HoverEpic committed:
Add ridering cache to CachedEntity

HoverEpic committed:
Add player input packet & translator

HoverEpic committed:
Fix Entity link packet (bad order & dava values)

HoverEpic committed:
Fix dismout vehicle

HoverEpic committed:
Register player input packet translator

HoverEpic committed:
Fix mount and dismount

HoverEpic committed:
WIP player input

lukeeey commented:
When compiling this branch i get an error:

```
java.lang.ClassNotFoundException: org.apache.maven.cli.MavenCli
        at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass(SelfFirstStrategy.java:50)
        at org.codehaus.plexus.classworlds.realm.ClassRealm.unsynchronizedLoadClass(ClassRealm.java:271)
        at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:247)
        at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:239)
        at org.codehaus.plexus.classworlds.launcher.Launcher.getMainClass(Launcher.java:144)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:266)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at **org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)**
```

HoverEpic commented:
What version of maven do you use? Can you try the latest?

lukeeey commented:
Its on my PC and im not at my PC rn so idk

lukeeey reviewed (approved):
Looks alright to me

tr7zw commented:
I just tried the Indev branch, and as far as I can see it has tons of bugs compared to master?
Item translation is wrong, some Entities get placed at the wrong location, some blocks are rotated worse than in master and the connection desyncs nearly instantly. Does the Project have a Discord/IRC, because I feel like I'm missing something?
